src/docfx/build/Build.cs

```diff
@@ -165,9 +165,9 @@ void ValidateBookmarks()
                 if (model != null && !hasErrors)
                 {
                     if (model is string str)
-                        context.WriteText(str, file.OutputPath);
```

Comment:
How about still putting `Docset.OutputProvider.GetOutputPath(this)` to `file.OutputPath`?
Then we do not need to change all these since the only thing we need is to extract the `OutputPath` logic for later change.